src/ralph/sim_cards/models.py

```diff
@@ -157,6 +165,101 @@ def assign_user(cls, instances, **kwargs):
         for instance in instances:
             instance.user = user
 
+    @classmethod
+    @transition_action(
+        form_fields={
+            'accept': {
+                'field': forms.BooleanField(
+                    label=_(
+                        'I have read and fully understand and '
+                        'accept the agreement.'
+                    )
+                )
+            },
+        }
+    )
+    def accept_asset_release_agreement(cls, instances, requester, **kwargs):
+        pass
+
+    @classmethod
+    def _generate_report(cls, name, requester, instances, language):
```

Comment:
DRY - extract `_generate_report` to common function and use it in `back_office/model.py` and here.

Reply:
Changed. And I push with changes.